src/canisters/frontend/ic-certified-assets/src/state_machine.rs

```diff
@@ -39,13 +40,205 @@ pub struct AssetEncoding {
     pub sha256: [u8; 32],
 }
 
+#[derive(Deserialize, CandidType, Serialize, Debug, Default, Clone, PartialEq, Eq)]
+pub struct RedirectUrl {
+    pub(crate) host: Option<String>,
+    pub(crate) path: Option<String>,
```

Comment:
Regex objects are generally not cheap to construct, but once built, they can be used:
```suggestion
    pub(crate) host: Option<Regex>,
    pub(crate) path: Option<Regex>,
```